### PR TITLE
feat: Add Google Tag Manager scripts for analytics

### PR DESCRIPTION
**Pull request by arkadyan**
Asana ticket: app portion of [Add Google analytics](https://app.asana.com/0/1204819229687089/1205784855883890)

This follows up on the [infrastructure update](https://github.com/mbta/devops/pull/1299).

Get the GTM ID from an environment variable.

Live now on [dev-orange](https://alerts-viewer-dev-orange.mbtace.com).

### CODE REVIEW

**Comment by arkadyan** [comment]
@paulswartz Since Erin is out this week, any chance you have a spare moment to review this? I'm happy to ask someone else if you are very busy at the moment.

File: lib/alerts_viewer_web/components/layouts/root.html.heex [hunk 1]
```diff
@@ -7,11 +7,32 @@
...
+          new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+          j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+          'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+          })(window,document,'script','dataLayer','<%= google_tag_manager_id() %>');
+    </script>
+    <!-- End Google Tag Manager -->
   </head>
   <body class="bg-white antialiased">
     <%= @inner_content %>
+    <!-- Google Tag Manager (noscript) -->
```
Comment (paulswartz) [inline lib/alerts_viewer_web/components/layouts/root.html.heex:26]: question: same question about the comment here.

File: lib/alerts_viewer_web/components/layouts/root.html.heex [hunk 2]
```diff
@@ -7,11 +7,32 @@
     <.live_title suffix=" · Alerts Viewer">
       <%= assigns[:page_title] || "Alerts Viewer" %>
     </.live_title>
+
     <link phx-track-static rel="stylesheet" href={~p"/assets/app.css"} />
     <script defer phx-track-static type="text/javascript" src={~p"/assets/app.js"}>
     </script>
+    <!-- Google Tag Manager -->
```
[the remark this answers is not in the document]
Comment (arkadyan) [reply lib/alerts_viewer_web/components/layouts/root.html.heex]: Yep, good point.

File: config/config.exs [hunk 3]
```diff
@@ -35,7 +35,8 @@ config :alerts_viewer,
     AlertsViewer.StopRecommendationAlgorithm.HeadwayDeviationComponent,
     AlertsViewer.StopRecommendationAlgorithm.AdherenceComponent
   ],
-  trip_updates_url: {:system, "TRIP_UPDATES_URL"}
+  trip_updates_url: {:system, "TRIP_UPDATES_URL"},
+  google_tag_manager_id: {:system, "GOOGLE_TAG_MANAGER_ID"}
```
[the remark this answers is not in the document]
Comment (arkadyan) [reply config/config.exs:39]: Thanks, good point. That should be a relatively quick update, but to keep things separate, I made a [ticket](https://app.asana.com/0/1167196461144361/1205799090178641) specifically for that work.

**Review by paulswartz — APPROVED** [review]

**Pull request merged**